src/main/java/org/myrobotlab/service/interfaces/HttpDataListener.java

```diff
@@ -1,8 +1,9 @@
 package org.myrobotlab.service.interfaces;
 
+import org.myrobotlab.framework.interfaces.NameProvider;
 import org.myrobotlab.service.data.HttpData;
 
-public interface HttpDataListener {
+public interface HttpDataListener extends NameProvider {

```

Comment:
if NameProvider is an interface.. this class should implement it .. not extend it.